air/src/stack/op_flags/mod.rs

```diff
@@ -0,0 +1,642 @@
+use super::EvaluationFrame;
+use crate::utils::binary_not;
+use vm_core::{
+    decoder::{NUM_OP_BITS, OP_BITS_RANGE, OP_BIT_EXTRA_COL_IDX},
+    Felt, FieldElement, DECODER_TRACE_OFFSET, TRACE_WIDTH, ZERO,
+};
+
+#[cfg(test)]
+pub mod tests;
+
+// CONSTANTS
+// ==================================================================================================
+
+pub const NUM_DEGREE_7_OPS: usize = 64;
+pub const NUM_DEGREE_6_OPS: usize = 16;
+pub const NUM_DEGREE_4_OPS: usize = 8;
+
+pub struct OpFlags<E: FieldElement> {
+    degree7_op_flags: [E; NUM_DEGREE_7_OPS],
+    degree6_op_flags: [E; NUM_DEGREE_6_OPS],
+    degree4_op_flags: [E; NUM_DEGREE_4_OPS],
+}
+
+#[allow(dead_code)]
+impl<E: FieldElement> OpFlags<E> {
+    // CONSTRUCTOR
+    // =================================================================================================
+
+    /// Returns a new instance of [OpFlags] instantiated with all stack operation flags.
+    pub fn new(frame: EvaluationFrame<E>) -> Self {
+        let (degree_6_op_flags, degree_4_op_flags) = set_op_flags_degree_4_and_6(&frame);
+
+        Self {
+            degree7_op_flags: set_op_flags_degree_7(&frame),
+            degree6_op_flags: degree_6_op_flags,
+            degree4_op_flags: degree_4_op_flags,
+        }
+    }
+
+    // OP FLAGS
+    // ================================================================================================
+
+    // ------ Degree 7 operations with no shift -------------------------------------------------
+
+    #[inline(always)]
+    fn noop(&self) -> E {
+        self.degree7_op_flags[0]
```

Comment:
I wonder whether instead of using a constant to index into an array we could compute the index using a `const fn` (so that the evaluation would be done at compile time) based on opcode (we'll need to change `Opeartion::op_code()` into a `const fn` too).

If the above works, we could rewrite these as something like:
```Rust
self.degree7_op_flags[get_op_index(Operation::Noop.op_code())]
```
And then we won't need to remember to update these mappings in case opcodes of operations change.

Reply:
I've created a separate function as per suggestion which accepts an opcode as input and returns the index at which the operation flag would be present in the respective flag array.